iceoryx_hoofs/include/iceoryx_hoofs/internal/design_pattern/polymorphic_handler.inl

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_HOOFS_DESIGN_PATTERN_POLYMORPHIC_HANDLER_INL
+#define IOX_HOOFS_DESIGN_PATTERN_POLYMORPHIC_HANDLER_INL
+
+#include "iceoryx_hoofs/design_pattern/polymorphic_handler.hpp"
+
+namespace iox
+{
+namespace design_pattern
+{
+
+namespace detail
+{
+
+template <typename Interface>
+void DefaultHooks<Interface>::onSetAfterFinalize(Interface&, Interface&) noexcept
+{
+    // we should not use an error handling construct (e.g. some IOX_ASSERT) here for dependency reasons
+    // we could in principle do nothing by default as well, but the misuse failure should have visible consequences
+    std::terminate();
```

Comment:
If possible, I wouldn't call terminate here. It is okay for now but ideally something like your proposed `FATAL`/`PANIC` should be called here once available

Reply:
We need to be careful with dependencies. I can change it to a do nothing implementation now.